src/sentry/rules/processing/delayed_processing.py

```diff
@@ -146,20 +146,30 @@ def get_rules_to_fire(
 ) -> DefaultDict[Rule, set[int]]:
     rules_to_fire = defaultdict(set)
     for alert_rule, slow_conditions in rule_to_slow_conditions.items():
-        for slow_condition in slow_conditions:
-            if slow_condition:
-                condition_id = slow_condition.get("id")
-                condition_interval = slow_condition.get("interval")
-                target_value = int(str(slow_condition.get("value")))
-                for condition_data, results in condition_group_results.items():
-                    if (
-                        alert_rule.environment_id == condition_data.environment_id
-                        and condition_id == condition_data.cls_id
-                        and condition_interval == condition_data.interval
-                    ):
-                        for group_id in rules_to_groups[alert_rule.id]:
-                            if results[group_id] > target_value:
+        action_match = alert_rule.data.get("action_match", "any")
+        for group_id in rules_to_groups[alert_rule.id]:
+            conditions_matched = 0
+            for slow_condition in slow_conditions:
+                if slow_condition:
+                    unique_condition = UniqueCondition(
+                        str(slow_condition.get("id")),
+                        str(slow_condition.get("interval")),
+                        alert_rule.environment_id,
+                    )
+                    results = condition_group_results.get(unique_condition, {})
+                    if results:
+                        target_value = int(str(slow_condition.get("value")))
+                        if results[group_id] >= target_value:
```

Comment:
Btw I put `>=` here but I don't remember what we actually do in the conditions, so we should confirm that

Reply:
Looks like just `>` https://github.com/getsentry/sentry/blob/master/src/sentry/rules/conditions/event_frequency.py#L155 so I'll update that